common/src/main/java/tds/common/web/interceptors/RestTemplateLoggingInterceptor.java

```diff
@@ -0,0 +1,88 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpRequest;
+import org.springframework.http.MediaType;
+import org.springframework.http.client.ClientHttpRequestExecution;
+import org.springframework.http.client.ClientHttpRequestInterceptor;
+import org.springframework.http.client.ClientHttpResponse;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+/**
+ * An interceptor for logging REST requests and responses
+ */
+public class RestTemplateLoggingInterceptor implements ClientHttpRequestInterceptor {
+    private static final Logger log = LoggerFactory.getLogger(RestTemplateLoggingInterceptor.class);
+    private static final String CONTENT_TYPE_SUBTYPE_JSON = "json";
+    private ObjectMapper objectMapper;
+
+    public RestTemplateLoggingInterceptor(ObjectMapper objectMapper) {
+        this.objectMapper = objectMapper;
+    }
+
+    @Override
+    public ClientHttpResponse intercept(final HttpRequest request, final byte[] body, final ClientHttpRequestExecution clientHttpRequestExecution) throws IOException {
+        logRequest(request, body);
+        ClientHttpResponse response = clientHttpRequestExecution.execute(request, body);
+        logResponse(response);
+        return response;
+    }
+
+    private void logRequest(final HttpRequest request, final byte[] body) {
+        final HttpHeaders headers = request.getHeaders();
+        String bodyString = new String(body, Charsets.UTF_8);
+
+        if (!bodyString.isEmpty() && headers.getAccept().contains(MediaType.APPLICATION_JSON)) {
+            try {
+                final Object json = objectMapper.readValue(bodyString, Object.class);
+                bodyString = objectMapper.writerWithDefaultPrettyPrinter().writeValueAsString(json);
+            } catch (IOException e) {
+                log.debug("Unable to parse the request as JSON. Printing raw request body string...");
+            }
+        }
+
+        log.debug("=========================================* REQUEST *==========================================");
+        log.debug(" Method/URI  :   {} - {}", request.getMethod(), request.getURI());
+        log.debug(" Headers     :   {} ", request.getHeaders());
+        log.debug(" Body        :   {} ", bodyString);
+        log.debug("==============================================================================================");
+    }
+
+    private void logResponse(final ClientHttpResponse response) throws IOException {
+        StringBuilder inputStringBuilder = new StringBuilder();
+        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(response.getBody(), Charsets.UTF_8));
+        String line = bufferedReader.readLine();
+
+        while (line != null) {
+            inputStringBuilder.append(line);
+            inputStringBuilder.append('\n');
+            line = bufferedReader.readLine();
```

Comment:
This feels like a lot of work to parse an InputStream as a string (and it doesn't look like the stream is being closed.)  I know we don't have apache commons' `IOUtils.toString(InputStream in)` available, but we could do something similar with Guava:
```
String bodyString = null;
try (final InputStream in = response.getBody()) {
  bodyString = CharStreams.toString(new InputStreamReader(in, Charsets.UTF_8));
}
```